base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
`{ } : Type` but `EmptyRow : Row`. Slide 20 is good at untangling this https://www.google.se/url?q=https://www.cs.cmu.edu/~neelk/rows.pdf&sa=U&ved=0ahUKEwi_mIb61-LOAhXIkCwKHf-9BxAQFggOMAE&usg=AFQjCNHgZPdKT__UjF9Zr9Yr3Ax5FF2hog